epregressions/diffs/ci_compare_script.py

```diff
@@ -257,55 +257,40 @@ def main_function(file_name, base_dir, mod_dir, base_sha, mod_sha, make_public,
                 # print("Processing output file: {0}, uploading to: {1}".format(filepath_to_send, filepath))
 
                 key = boto.s3.key.Key(bucket, file_path)
-                file_to_send = open(file_path_to_send, 'r')
-                key.set_contents_from_string(file_to_send.read())
-
-                if make_public:
-                    key.make_public()
-
-                htmlkey = boto.s3.key.Key(bucket, file_path + ".html")
-                htmlkey.set_contents_from_string("""
+                with open(file_path_to_send, 'r') as file_to_send:
+                    contents = file_to_send.read()
+                    key.set_contents_from_string(contents)
+
+                    if make_public:
+                        key.make_public()
+
+                    htmlkey = boto.s3.key.Key(bucket, file_path + ".html")
+
+                    if file_path_to_send.endswith('.htm'):
```

Comment:
If we are creating an htm diff, just put the raw htm contents and mark the content type and disposition properly so it shows up as a nice html page right on CI.